uwblueprint/don-efficace
# added endpoint to create a Stripe Customer

## Notion ticket link

[create Stripe endpoint to create customer](https://www.notion.so/uwblueprintexecs/create-Stripe-endpoint-to-create-customer-1b662427203c4741a57a1378a25d4136?pvs=4)



## Implementation description
* added a new route /create-customer that calls a service to create a customer
* added a stripe Service called createCustomer that takes in a name, email and a payment method to create a customer
* added validators to check that the input to the endpoint is valid
* created a frontend Component to test the endpoint and it successfully creates a Customer object if valid info is passed and produces errors otherwise



## Steps to test
1. Import the frontend component and add a route into App.tsx 
2. create a new route constant in Routes.ts
3. run the project locally and go to the route localhost:3000/stripe-customer (or whatever you defined the route constant as)
4. when entering the card number, enter "4242 4242 ..."

Once you enter the info into the form, you will see:
![image](https://github.com/user-attachments/assets/80f59a79-6d8f-4808-be3a-eb28af81d434)

sample payment object:
![image](https://github.com/user-attachments/assets/596aa4c1-7482-4c54-a944-2e6ec030bea5)



## What should reviewers focus on?
* valid implementation and logic of the customer object (correct parameters?)


## Checklist
